### ISSUE TITLE
Yolov5s6 performance during training

### ISSUE THREAD

**Issue by thecastlingking** [ultralytics/yolov5#6518]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hey @glenn-jocher 

I have been using your network to detect avocados and its gone well so far, but the **yolov5s6.pt** model **dropped** during training but only for **one specific image size**, **512x512**, but **worked fine with 448x448**. I ran it again thinking it was a glitch, but it happened again. 
(see images, **green line = yolov5s6.pt** mAP)

I was wondering what the cause might be so I would be able to justify what's going on. Any ideas? 

I've also included my model and runtime code to benefit any solution. 
(Let me know if I need to include anything else as I don't really post much on GitHub)

**First run**
![image](https://user-images.githubusercontent.com/64767196/152275728-d36d0731-a682-4045-9c2b-b0a86cd70083.png)

**Second run**
![image](https://user-images.githubusercontent.com/64767196/152275744-47e56146-03e1-44b1-912b-b0ac3ca5f8d7.png)

### Additional

python train.py --img 512 --batch 32 --epochs 200 --data ./data.yaml --weights '' --cfg ./models/yolov5s6.yaml --name yolov5s6_results
train: weights='', cfg=./models/yolov5s6.yaml, data=./data.yaml, hyp=data\hyps\hyp.scratch.yaml, epochs=200, batch_size=32, imgsz=512, rect=False, resume=False, nosave=False, noval=False, noautoanchor=False, evolve=None, bucket=, cache=None, image_weights=False, device=, multi_scale=False, single_cls=False, adam=False, sync_bn=False, workers=8, project=runs\train, name=yolov5s6_results, exist_ok=False, quad=False, linear_lr=False, label_smoothing=0.0, patience=100, freeze=0, save_period=-1, local_rank=-1, entity=None, upload_dataset=False, bbox_interval=-1, artifact_alias=latest
YOLOv5  v6.0-124-g1075488 torch 1.10.0 CUDA:0 (Quadro P2200, 5120MiB)

hyperparameters: lr0=0.01, lrf=0.1, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=0.05, cls=0.5, cls_pw=1.0, obj=1.0, obj_pw=1.0, iou_t=0.2, anchor_t=4.0, fl_gamma=0.0, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0
Weights & Biases: run 'pip install wandb' to automatically track and visualize YOLOv5  runs (RECOMMENDED)
TensorBoard: Start with 'tensorboard --logdir runs\train', view at http://localhost:6006/

Overriding model.yaml nc=80 with nc=1
                from  n    params  module                                           arguments
  0                -1  1      3520  models.common.Conv                      [3, 32, 6, 2, 2]
  1                -1  1     18560  models.common.Conv                      [32, 64, 3, 2]
  2                -1  1     18816  models.common.C3                        [64, 64, 1]
  3                -1  1     73984  models.common.Conv                      [64, 128, 3, 2]
  4                -1  2    115712  models.common.C3                        [128, 128, 2]
  5                -1  1    295424  models.common.Conv                      [128, 256, 3, 2]
  6                -1  3    625152  models.common.C3                        [256, 256, 3]
  7                -1  1    885504  models.common.Conv                      [256, 384, 3, 2]
  8                -1  1    665856  models.common.C3                        [384, 384, 1]
  9                -1  1   1770496  models.common.Conv                      [384, 512, 3, 2]
 10                -1  1   1182720  models.common.C3                        [512, 512, 1]
 11                -1  1    656896  models.common.SPPF                      [512, 512, 5]
 12                -1  1    197376  models.common.Conv                      [512, 384, 1, 1]
 13                -1  1         0  torch.nn.modules.upsampling.Upsample    [None, 2, 'nearest']
 14           [-1, 8]  1         0  models.common.Concat                    [1]
 15                -1  1    813312  models.common.C3                        [768, 384, 1, False]
 16                -1  1     98816  models.common.Conv                      [384, 256, 1, 1]
 17                -1  1         0  torch.nn.modules.upsampling.Upsample    [None, 2, 'nearest']
 18           [-1, 6]  1         0  models.common.Concat                    [1]
 19                -1  1    361984  models.common.C3                        [512, 256, 1, False]
 20                -1  1     33024  models.common.Conv                      [256, 128, 1, 1]
 21                -1  1         0  torch.nn.modules.upsampling.Upsample    [None, 2, 'nearest']
 22           [-1, 4]  1         0  models.common.Concat                    [1]
 23                -1  1     90880  models.common.C3                        [256, 128, 1, False]
 24                -1  1    147712  models.common.Conv                      [128, 128, 3, 2]
 25          [-1, 20]  1         0  models.common.Concat                    [1]
 26                -1  1    296448  models.common.C3                        [256, 256, 1, False]
 27                -1  1    590336  models.common.Conv                      [256, 256, 3, 2]
 28          [-1, 16]  1         0  models.common.Concat                    [1]
 29                -1  1    715008  models.common.C3                        [512, 384, 1, False]
 30                -1  1   1327872  models.common.Conv                      [384, 384, 3, 2]
 31          [-1, 12]  1         0  models.common.Concat                    [1]
 32                -1  1   1313792  models.common.C3                        [768, 512, 1, False]
 33  [23, 26, 29, 32]  1     23112  models.yolo.Detect                      [1, [[19, 27, 44, 40, 38, 94], [96, 68, 86, 152, 180, 137], [140, 301, 303, 264, 238, 542], [436, 615, 739, 380, 925, 792]], [128, 256, 384, 512]]
Model Summary: 355 layers, 12322312 parameters, 12322312 gradients, 16.2 GFLOPs

Scaled weight_decay = 0.0005
optimizer: SGD with parameter groups 75 weight, 79 weight (no decay), 79 bias
train: Scanning '..\train\labels' images and labels...3516 found, 0 missing, 0 empty, 0 corrupted: 100%|████████████████████████████████| 3516/3516 [00:06<00:00, 584.04it/s]
train: New cache created: ..\train\labels.cache
val: Scanning '..\valid\labels' images and labels...504 found, 0 missing, 0 empty, 0 corrupted: 100%|██████████████████████████████████████| 504/504 [00:12<00:00, 39.99it/s]
val: New cache created: ..\valid\labels.cache
module 'signal' has no attribute 'SIGALRM'

AutoAnchor: 3.11 anchors/target, 0.981 Best Possible Recall (BPR). Current anchors are a good fit to dataset
Image sizes 512 train, 512 val
Using 8 dataloader workers
Logging results to runs\train\yolov5s6_results
Starting training for 200 epochs...

**Comment by glenn-jocher** [comment]
@thecastlingking 👋 Hello! Thanks for asking about **training reproducibility**. YOLOv5 🚀 uses a single training seed which is set here using the `init_seeds()` function.
https://github.com/ultralytics/yolov5/blob/6865d19a92d8c160c7fc3c92256627dadce1cd1e/train.py#L108

This function sets `python`, `numpy` and `torch` seeds and updates `cudnn` settings:
https://github.com/ultralytics/yolov5/blob/6865d19a92d8c160c7fc3c92256627dadce1cd1e/utils/general.py#L122-L130

To set a new seed for example:
```python
    init_seeds(2 + RANK)  # set seed=2
```

Note that even when using the same seed trainings may produce different results, especially when using CUDA backends. See https://pytorch.org/docs/stable/notes/randomness.html for details on factors affecting PyTorch training reproducibility and sources of randomness.

<img width="949" alt="Screen Shot 2022-01-08 at 11 20 59 AM" src="https://user-images.githubusercontent.com/26833433/148660366-9155dfd9-351c-4174-818f-2b1b907b3e4e.png">

Good luck 🍀 and let us know if you have any other questions!
